### ISSUE TITLE
PSNR Not reached 48?

### ISSUE THREAD

**Issue by yuanhaowang1213** [aasharma90/Unprocessing_for_RAW_Denoising#4]
Hi, 
I have tried with your model, but the PSNR is around 40.4856 on RAW. But it could reach 48.49 on the paper? So what happened?

 | Uses VST | Denoised on | PSNR on raw | SSIM on raw | PSNR on sRGB | SSIM on sRGB | Details
 | -- | -- | -- | -- | -- | -- | --
 | 0 | raw | 40.4856 | 0.8581 | 31.4927 | 0.696

I also tried with the parameter with the Flicker dataset 25k, but the loss stop at 
0.01108. So should I add more epoches?

**Comment by aasharma90** [comment]
HI @yuanhaowang1213, 

Have you tried evaluating the official tensorflow code, i.e. whether it reaches the same/similar performance as reported in the paper? I'm not sure whether any official pre-trained tensorflow model is available, but if there is one, you should try to evaluate that first. Then, maybe you should try to train-from-scratch the official tensorflow code and see what is the outcome. The code here basically is a PyTorch implementation of the original tensorflow code, and thus unless we know how the official pre-trained and trained-from-scratch  tensorflow models are performing, it would be difficult to understand the reason of the performance gap that you've reported. 

Given that the quantitative evaluation can be only done through official submission on the DND webpage, I couldn't do any comparisons mentioned above. If you have the results, you can let me know, and I can have a look into it.

**Comment by yuanhaowang1213** [comment]
Hi, @aasharma90 

The pre-trianed model is available on line.

Here attached the result by evaluating the official RGB model. I will try with the train-from-scratch, and make more results.


| Denoised on | PSNR on raw | SSIM on raw | PSNR on sRGB | SSIM on sRGB | Details
| -- | -- | -- | -- | -- | --
| 0 | raw | 48.8905 | 0.9823 | 40.1728 | 0.9604

This is with RAW and quite close to the result on the paper.

I added this code to line 152 to export the mat file needed.

      Img_denoised_out = raw_image_out.detach().cpu()
      Idenoised_crop = noisy_crop.copy()
      for yy in range(2):
        for xx in range(2):
          Idenoised_crop[yy:height:2, xx:width:2] = np.float32(Img_denoised_out[0, 2*yy + xx, :, :])
      save_file = os.path.join(args.results_dir , testname,'%04d_%02d.mat'%(i+1, k+1))
      sio.savemat(save_file, {'Idenoised_crop': Idenoised_crop})

**Comment by yuanhaowang1213** [comment]
Hi, @aasharma90 

I have done more test with the code. And found that I made a mistake for the output. The output should be the flipped image, however, I tried with the raw_image_out data. 

Sorry to make this mistake.

The result with checkpoint400.rar is 

| Denoised on | PSNR on raw | SSIM on raw | PSNR on sRGB | SSIM on sRGB | Details
| -- | -- | -- | -- | -- | --
|0 |raw | 47.9167 | 0.9785 | 39.2103 | 0.9473
